src/client/java/dev/creesch/model/WebsocketMessageBuilder.java

```diff
@@ -0,0 +1,74 @@
+package dev.creesch.model;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import dev.creesch.WebInterface;
+import dev.creesch.util.MinecraftServerIdentifier;
+import dev.creesch.util.NamedLogger;
+import net.minecraft.SharedConstants;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.text.Text;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.UUID;
+
+public class WebsocketMessageBuilder {
+    private static final Gson gson = new Gson();
+
+    /**
+     * Processes both chat and game messages, converting them to the appropriate format
+     *
+     * @param message The Minecraft text message to process
+     * @param client The Minecraft client instance
+     */
+    public static WebsocketJsonMessage createChatMessage(Text message, MinecraftClient client) {
+        if (client.world == null) {
+            throw new MessageBuildException("Cannot create chat message: client world is null");
+        }
+
+        // Can't use GSON for Text serialization easily, using Minecraft's own serializer.
+        String minecraftChatJson = Text.Serialization.toJsonString(message, client.world.getRegistryManager());
+
+        // Explicitly use UTC time for consistency across different timezones
+        long timestamp = Instant.now(Clock.systemUTC()).toEpochMilli();
+        WebsocketJsonMessage.ChatServerInfo serverInfo = MinecraftServerIdentifier.getCurrentServerInfo();
+        String minecraftVersion = SharedConstants.getGameVersion().getName();
+        // UUID used to prevent duplicates when doing
+        String messageUUID = UUID.nameUUIDFromBytes((timestamp + minecraftChatJson).getBytes()).toString();
+
+        // Back to objects we go
+        JsonObject messageObject = new JsonObject();
```

Comment:
I am strongly considering making the chatMessage Payload into a specific object. Seems cleaner than just bolting on json properties here. Would also make use down the line slightly easier.

Reply:
In fact, that is what I am going to do.